app/src/main/java/org/fossasia/susi/ai/skills/groupwiseskills/GroupWiseSkillsFragment.kt

```diff
@@ -93,6 +96,11 @@ class GroupWiseSkillsFragment : Fragment(), IGroupWiseSkillsView, SwipeRefreshLa
         skillsAdapter.notifyDataSetChanged()
     }
 
+    override fun onAttach(context: Context) {
+        super.onAttach(context);
+        skillCallback = activity as SkillFragmentCallback
```

Comment:
Type unsafe and high crash hazard in future

Read about activity - fragment communication and how to do it safely

Reply:
@iamareebjamal I read about it. But, 'skillCallback' is a lateinit var of type SkillFragmentCallback interface. How should I initialise it without casting? I also need to pass this 'skillCallback' in the adapter. It would be helpful if you could please help me a bit with this.

Reply:
Yes that's true, but here it can crash with ClassCastException, handle the possible error case through an if statement

Reply:
Can I use try catch block to handle Class Cast exception?